sdk/go/common/resource/plugin/rpc_rapid_test.go

```diff
@@ -17,70 +17,298 @@ package plugin
+		return rapid.Just(stringValue(UnknownStringValue)).Draw(t, "unknowns").(*structpb.Value)
+	})
 }
 
-// Generates Output values.
-func outputGen() *rapid.Generator {
-	propertyValueG := propertyValueGen()
-	urnsG := rapid.SliceOf(urnGen())
-	return rapid.Custom(func(t *rapid.T) resource.Output {
-		element := propertyValueG.Draw(t, "element").(resource.PropertyValue)
-		known := rapid.Bool().Draw(t, "known").(bool)
-		secret := rapid.Bool().Draw(t, "secret").(bool)
-		deps := urnsG.Draw(t, "dependencies").([]resource.URN)
-		return resource.Output{
-			Element:      element,
-			Known:        known,
-			Secret:       secret,
-			Dependencies: deps,
+// NullGenerator generates the null resource.PropertyValue.
+func NullGenerator() *rapid.Generator {
+	return rapid.Custom(func(t *rapid.T) *structpb.Value {
+		return rapid.Just(&structpb.Value{
+			Kind: &structpb.Value_NullValue{
+				NullValue: structpb.NullValue_NULL_VALUE,
+			},
+		}).Draw(t, "nulls").(*structpb.Value)
+	})
+}
+
+// BoolGenerator generates boolean resource.PropertyValues.
+func BoolGenerator() *rapid.Generator {
+	return rapid.Custom(func(t *rapid.T) *structpb.Value {
+		return &structpb.Value{
+			Kind: &structpb.Value_BoolValue{
+				BoolValue: rapid.Bool().Draw(t, "booleans").(bool),
+			},
 		}
 	})
 }
 
-func normOutput(pv *resource.PropertyValue) {
-	if pv.IsOutput() {
-		out := pv.OutputValue()
-		if len(out.Dependencies) == 0 && out.Dependencies != nil {
-			out.Dependencies = nil
+// NumberGenerator generates numeric resource.PropertyValues.
+func NumberGenerator() *rapid.Generator {
+	return rapid.Custom(func(t *rapid.T) *structpb.Value {
+		return &structpb.Value{
+			Kind: &structpb.Value_NumberValue{
+				NumberValue: rapid.Float64().Draw(t, "numbers").(float64),
+			},
 		}
-		pv.V = out
-	}
+	})
 }
 
-func TestOutputValueTurnaround(t *testing.T) {
-	rapid.Check(t, func(t *rapid.T) {
-		out := outputGen().Draw(t, "output").(resource.Output)
-		v := resource.NewOutputProperty(out)
+// StringGenerator generates string resource.PropertyValues.
+func StringGenerator() *rapid.Generator {
+	return rapid.Custom(func(t *rapid.T) *structpb.Value {
+		return stringValue(rapid.String().Draw(t, "strings").(string))
+	})
+}
+
+// TextAssetGenerator generates textual *resource.Asset values.
+func TextAssetGenerator() *rapid.Generator {
+	return rapid.Custom(func(t *rapid.T) *structpb.Value {
+		return &structpb.Value{
+			Kind: &structpb.Value_StructValue{
+				StructValue: &structpb.Struct{
+					Fields: map[string]*structpb.Value{
+						resource.SigKey:            stringValue(resource.AssetSig),
+						resource.AssetTextProperty: stringValue(rapid.String().Draw(t, "text asset contents").(string)),
+					},
+				},
+			},
+		}
+	})
+}
+
+// AssetGenerator generates *resource.Asset values.
+func AssetGenerator() *rapid.Generator {
+	return TextAssetGenerator()
+}
+
+// LiteralArchiveGenerator generates *resource.Archive values with literal archive contents.
```

Comment:
These aren't *resource.Archive values but *structpb.Value values right that represent the gRPC wire form of the *resource.Archive?

Reply:
Ah, oops--I meant to fix up these comments.